src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -118,6 +122,24 @@ public void setPersons(List<Person> persons) {
         internalList.setAll(persons);
     }
 
+    /**
+     * Filters this list according to the {@code predicate} given.
+     * {@code predicate} must not be null.
+     */
+    public FilteredList<Person> filterPersons(Predicate<Person> predicate) {
+        requireAllNonNull(predicate);
+        return internalList.filtered(predicate);
+    }
+
+    /**
+     * Sorts this list according to the {@code comparator} given.
+     * {@code comparator} must not be null.
+     */
+    public SortedList<Person> sortPersons(Comparator<Person> comparator) {
+        requireAllNonNull(comparator);
+        return internalList.sorted(comparator);
+    }
+
```

Comment:
I will delete filtering as it is not used. But I can't think of a way to do sorting without modifying the `UniquePersonList`. Any suggestions?

Reply:
what i mean:
![image](https://user-images.githubusercontent.com/72195240/159958881-839dabc2-d31b-40ec-9a63-b74d44a29c32.png)

Reply:
Within the modelManager, wrap the filteredList in a [SortedList](https://docs.oracle.com/javase/8/javafx/api/javafx/collections/transformation/SortedList.html#SortedList-javafx.collections.ObservableList-). Since the `filteredList` is a subtype of `ObservableList`, this is possible.

Then, if we want to sort, just set the comparator to the `SortedList`.
If we want to filter, set the predicate to the underlying `filteredList` that we just wrapped.

Reply:
but `filteredPersons` is of type `filteredList`. How can it be a `sortedList`?

Reply:
Thank you. I thought the predicate won't be able to apply in the `SortedList`, but it does. I've updated the implementation.